Upgrade @elastic/eui to 85.0.1

tkajtoch opened the pull request:
## Summary

`eui@84.0.0` ⏩ `eui@85.0.1`

## [`85.0.1`](https://github.com/elastic/eui/tree/v85.0.1)

**Bug fixes**

- Fixed `EuiFilterGroup`'s responsive styles ([#6983](https://github.com/elastic/eui/pull/6983))

## [`85.0.0`](https://github.com/elastic/eui/tree/v85.0.0)

- Updated `EuiThemeProvider` to set an Emotion theme context that returns the values of `useEuiTheme()` ([#6913](https://github.com/elastic/eui/pull/6913))
- Added `size` prop to `EuiStepsHorizontal`, defaulting to the previous size of `m` ([#6928](https://github.com/elastic/eui/pull/6928))
- Added new `s` sizing to `EuiStepsHorizontal` ([#6928](https://github.com/elastic/eui/pull/6928))
- Added `at` and `key` icon glyphs. ([#6934](https://github.com/elastic/eui/pull/6934))
- Added a new `cloneElementWithCss` Emotion utility ([#6939](https://github.com/elastic/eui/pull/6939))
- Updated `EuiPopover` to allow consumer control of all `focusTrapProps` ([#6955](https://github.com/elastic/eui/pull/6955))

**Bug fixes**

- Fixed `EuiDataGrid` height calculation bug when browser zoom levels are not 100% ([#6895](https://github.com/elastic/eui/pull/6895))
- Fixed `EuiTab` not correctly passing selection color state to `prepend` and `append` children ([#6938](https://github.com/elastic/eui/pull/6938))
- Fixed `EuiInputPopover` to allow consumer control of its focus trap via `focusTrapProps` ([#6955](https://github.com/elastic/eui/pull/6955))

**Breaking changes**

- `EuiProvider` will no longer render multiple or duplicate nested instances of itself. If a nested `EuiProvider` is detected, that instance will return early without further processing, and will warn if configured to do so via `setEuiDevProviderWarning`. For nested theming, use `EuiThemeProvider` instead. ([#6949](https://github.com/elastic/eui/pull/6949))
- Removed `onTrapDeactivation` prop from `EuiPopover`. Use `focusTrapProps.onDeactivation` instead ([#6955](https://github.com/elastic/eui/pull/6955))

**CSS-in-JS conversions**

- Converted `EuiFilterGroup` and `EuiFilterButton` to Emotion; Removed styles attached to `.euiFilterGroup__popoverPanel` ([#6957](https://github.com/elastic/eui/pull/6957))

tkajtoch committed:
Upgrade @elastic/eui to 85.0.0

cee-chen committed:
[Presentation] Remove `.euiFilterGroup__popoverPanel` usage

- `.presFilterByType__panel` already has its own width Sass, so move the width CSS there instead

cee-chen committed:
[EuiFilterGroup] Remove all other references to `.euiFilterGroup__popoverPanel`

- After Emotion conversion, no longer has any styles attached to it

- use Emotion CSS instead

tkajtoch committed:
increase EuiFilterButton selector specificity in failing unit tests due to emotion conversion

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

Heenawter reviewed (approved):
Presentation changes LGTM :+1:

cee-chen committed:
Update EUI to v85.0.1

jughosta commented on packages/kbn-unified-field-list/src/components/field_list_filters/field_type_filter.tsx:
```diff
@@ -149,7 +149,7 @@ export function FieldTypeFilter<T extends FieldListItem = DataViewField>({
     <EuiPopover
       id="unifiedFieldTypeFilter"
-      panelClassName="euiFilterGroup__popoverPanel"
+      panelProps={{ css: { width: euiTheme.base * 18 } }}
```
In Discover, the filter button width and the filter item height look larger than it was in 8.9:

Before:
<img width="400" alt="Screenshot 2023-07-25 at 12 08 48" src="https://github.com/elastic/kibana/assets/1415710/dfd670c2-645a-4e44-81d8-2d7e6bcd2f9c">

Now:
<img width="400" alt="Screenshot 2023-07-25 at 12 08 58" src="https://github.com/elastic/kibana/assets/1415710/1ce23f6d-556f-4417-8abf-ba9ab9643352">

tkajtoch replied:
Thanks @jughosta I'll look into that!

cee-chen replied:
The filter button width should be fixed as with https://github.com/elastic/kibana/pull/162209/commits/14e74692722afe0ebb981bf812e6f87a267fbaf4 - I hadn't realized folks in Kibana were using `EuiFilterButton` outside of `EuiFilterGroup`, so that's my bad :)

Regarding the filter item height - this looks like it was affected by https://github.com/elastic/kibana/pull/161592 (Sass styling order). I pushed up a specificity override for now (https://github.com/elastic/kibana/pull/162209/commits/f5a55e0c537bf72255c04dfc9bca563d123a4d45) which we can remove once more of EUI is on Emotion 🤞 

Thanks for your eagle eye on these things @jughosta, you're amazing!!

jughosta replied:
Thanks for the fixes, @cee-chen! You are amazing! 😉

cee-chen committed:
[EuiFilterButton] Fix Kibana min-width overrides

- `min-width` is now set on `EuiFilterButtons` even outside of `EuiFilterGroup`s, so CSS needs to update to take that into account

cee-chen committed:
[EuiContextMenuItem] Fix specificity issue

clintandrewhall commented on packages/content-management/table_list_view_table/src/components/table_sort_select.tsx:
```diff
@@ -160,7 +160,7 @@ export function TableSortSelect({ tableSort, hasUpdatedAtMetadata, onChange }: P
       panelPaddingSize="none"
       anchorPosition="downCenter"
-      panelClassName="euiFilterGroup__popoverPanel"
+      panelProps={{ css: { width: euiTheme.base * 18 } }}
```
nit: would prefer `18` to be a constant.  Not sure what that number is supposed to correspond to. 

@ryankeairns, you like to call this out often, what do you think?

cee-chen replied:
Y'all are super welcome to set your own constant. The number is literally what `.euiFilterGroup__popoverPanel` was setting as a width in CSS/Sass ([link](https://github.com/elastic/eui/blob/2228bef71e9c855be4a27163d763df669d12d91f/src/components/filter_group/_filter_group.scss#L37-L41)), it's just not hidden behind an EUI CSS class anymore, as we are no longer providing component-specific CSS utilities to consumers.

In any case, `width` is fairly content-specific so we would prefer consumers figure out what width they want to display based on their content and not on a generic.

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v85.0.0

spong reviewed (approved):
Security Detection Rule Mgmt changes (just two test updates) LGTM! Thanks @tkajtoch! 🙂

jughosta reviewed (approved):
Data Discovery changes LGTM 👍

tkajtoch commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v85.0.0

tkajtoch commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v85.0.0

tkajtoch commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v85.0.0

ryankeairns reviewed (approved):
👍 based upon this reply: https://github.com/elastic/kibana/pull/162209/files#r1273854410

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v85.0.0

cee-chen commented:
@elastic/response-ops As a heads up, we'll be asking @elastic/kibana-operations for an admin merge today as this PR has been open for over a week. If you encounter an issues that you think may be related to the upgrade after the fact (knock on wood), please feel free to ping us.
